ui/src/app/pages/admin/admin-workspace.tsx

```diff
@@ -77,53 +88,121 @@ const formatMB = (fileSize: number): string => {
   }
 };
 
-const FileDetailsTable = (props: {data: Array<FileDetail>, bucket: string}) => {
-  const {data, bucket} = props;
+const parseLocation = (file: FileDetail, bucket: string): string => {
+  const prefixLength = bucket.length;
+  const start = prefixLength + 1;  // slash after bucket name
+  const suffixPos = file.path.lastIndexOf(file.name);
+  const end = suffixPos - 1;  // slash before filename
+
+  return file.path.substring(start, end);
+};
+
+const NOTEBOOKS_DIRECTORY = 'notebooks';
+const NOTEBOOKS_SUFFIX = '.ipynb';
+const MAX_NOTEBOOK_READ_SIZE_BYTES = 5 * 1000 * 1000; // see NotebooksServiceImpl
+
+interface NameCellProps {
+  file: FileDetail;
+  bucket: string;
+  workspaceNamespace: string;
+  accessReason: string;
+}
+
+const NameCell = (props: NameCellProps) => {
+  const {file, bucket, workspaceNamespace, accessReason} = props;
+  const filename = file.name.trim();
+
+  const filenameSpan = () => <span>{filename}</span>;
+
+  const fileTooLarge = () => <FlexRow>
+    {filenameSpan()}
+    <TooltipTrigger
+        content={`Files larger than ${formatMB(MAX_NOTEBOOK_READ_SIZE_BYTES)} MB are too large to preview`}
+    ><Button style={styles.previewButton} disabled={true}>Preview</Button>
+    </TooltipTrigger>
+  </FlexRow>;
+
+  const navigateToPreview = () => navigate(
+      ['admin', 'workspaces', workspaceNamespace, filename],
+      { queryParams: { accessReason: accessReason } });
+
+  const fileWithPreviewButton = () => <FlexRow>
+    {filenameSpan()}
+    <TooltipTrigger content='Please enter an access reason below' disabled={accessReason && accessReason.trim()}>
+      <Button style={styles.previewButton}
+              disabled={!accessReason || !accessReason.trim()}
+              onClick={navigateToPreview}>Preview</Button>
+    </TooltipTrigger>
+  </FlexRow>;
+
+  // remove first check after RW-5626
+  const isNotebook = () => (NOTEBOOKS_DIRECTORY === parseLocation(file, bucket)) && filename.endsWith(NOTEBOOKS_SUFFIX);
+  const isTooLargeNotebook = () => isNotebook() && (file.sizeInBytes > MAX_NOTEBOOK_READ_SIZE_BYTES);
+
+  // if (tooLarge()) fileTooLarge();
+  // else if (isNotebook()) fileWithPreviewButton();
+  // else filenameSpan();
```

Comment:
Are these comments needed?

Reply:
I find it difficult to find good documentation or examples for `fp`, so it's my hope that this will be a useful "translation" of the below code